xmtp_db/src/encrypted_store/client_events.rs

```diff
@@ -0,0 +1,93 @@
+use super::{
+    ConnectionExt, DbConnection, group::ConversationType, group_intent::IntentKind,
+    schema::client_events::dsl,
+};
+use crate::{StorageError, Store, impl_store, schema::client_events};
+use diesel::{Insertable, Queryable, prelude::*};
+use serde::{Deserialize, Serialize};
+use xmtp_common::{NS_IN_30_DAYS, time::now_ns};
+
+#[derive(Insertable, Queryable, Debug, Clone)]
+#[diesel(table_name = client_events)]
+#[diesel(primary_key(created_at_ns))]
+pub struct ClientEvents {
+    pub created_at_ns: i64,
+    pub details: serde_json::Value,
+}
+
+impl_store!(ClientEvents, client_events);
+
+impl ClientEvents {
+    pub fn track<C: ConnectionExt>(db: &DbConnection<C>, event: impl AsRef<ClientEvent>) {
```

Comment:
Are these used anywhere yet? 

It would be nice to have a configuration option on the client builder or somewhere where we can enable/disable the stats feature. Sort of like the ApiDebugWrapper for the ApiClient. It probably won't have a huge impact on performance, but it would be nice to have that safety switch that could be useful for benchmarks/etc.

Reply:
Yeah, great callout.  I thought about that last night too. I have a follow-up PR that will add a config.

We track in 3 places right now, client creation, welcomes, and queue intent. Just a start.